### ISSUE TITLE
Can you document tf metrics use in tf.keras?

### ISSUE THREAD

**Issue by bhack** [tensorflow/tensorflow#17168]
There is no official documentation on how to use tf metrics in tf.keras. 
Two stackoverflow reference but I don't know if it is a API stable solution: 
https://stackoverflow.com/questions/45947351/how-to-use-tensorflow-metrics-in-keras
https://stackoverflow.com/questions/43158719/how-can-i-use-tensorflow-metric-function-within-keras-models

**Comment by asimshankar** [comment]
@fchollet @anj-s for comment.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by fchollet** [comment]
This is generally not supported. But certain TF metrics can work with Keras.

A metrics function, in Keras, for the time being, is a stateless tensor function that takes as arguments `(y_true, y_pred)` (target tensor and prediction tensor). If you're looking at a stateless TF metric (a metric that does not create weight variables in the background when you call it), then you can wrap it to have this signature, in most cases. Then you can pass the wrapped function to Keras.

Any stateful metric, in Keras, should be implemented as a metrics layer. [Here is an example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/_impl/keras/metrics_test.py#L81-L119).

In general, we recommend using only metrics from `tf.keras.metrics` with tf.keras. Yes, we're aware it's a bad UX to have multiple incompatible metrics modules, and we'll fix it in the future.

**Comment by bhack** [comment]
@fchollet So is this the same conclusion comment for https://github.com/keras-team/keras/issues/6050? I think that unifying the API it could be great but it is not a short term solution. Many standard tensorflow metrics are not covered in Keras so I think that we need to document this in some ways also if the documentation will be "don't try this at home".
